api/notification.php

```diff
@@ -4,14 +4,15 @@
 require_once __DIR__ . "/../classes.php";
 
 $system = new System();
+$system->startTransaction();
```

Comment:
Huh, wonder if we should just add this call to the constructor method of System (with a parameter to disable it if desired)

Reply:
We might want to be explicit for visibility since we're being explicit with `$system->commit/rollbackTransaction`?

Reply:
Yeah, we could leave this for now. But we're pulling the commit into `API::exitWithData`. Maybe the way to go is to make an `API::init` method that returns the $system object with transaction started and `is_api_request` set to true

Reply:
That approach makes sense. So roll this functionality into the API class:
```
$system = new System();
$system->startTransaction();
$system->is_api_request = true;
return $system;
```
and call this at the start of each API?
`$system = API::init();`

Follow-up PR okay?

Reply:
Yup, sounds good!